src/root_finding/root_finding.jl

```diff
@@ -1,39 +1,127 @@
 
+# immutable Root{T<:Real}
```

Comment:
I really like the idea; have you benchmark it against previous behavior? I would guess it is much faster now, isn't it?